src/main/java/mods/eln/misc/FloatLinear.kt

```diff
@@ -0,0 +1,130 @@
+package mods.eln.misc
+
+data class Vec3f(var x: Float, var y: Float, var z: Float) {
```

Comment:
I would really like some comments here. I mean, yes, it's a 3-element vector class with linear operations, but the other classes down below are nowhere near as obvious.